packages/@angular/cli/commands/generate.ts

```diff
@@ -78,6 +78,11 @@ export default Command.extend({
         `The \`ng generate ${name}\` command requires a name to be specified.`);
     }
 
+    if (/^\d/.test(rawArgs[1])) {
+      SilentError.debugOrThrow('@angular/cli/commands/generate',
+          `The \`ng generate ${name} ${rawArgs[1]}\` file name shouldn't begin with a digit.`);
```

Comment:
Since this is a fatal error, "cannot" would probably be more appropriate then "shouldn't" which generally has a _softer_ meaning.